rpcs3/CMakeLists.txt

```diff
@@ -153,7 +153,9 @@ elseif(UNIX)
             ${CMAKE_SOURCE_DIR}/bin/git $<TARGET_FILE_DIR:rpcs3>/git)
 elseif(WIN32)
     add_custom_command(TARGET rpcs3 POST_BUILD
+        COMMAND "${CMAKE_COMMAND}" -E remove "${CMAKE_SOURCE_DIR}/bin/rpcs3.exe"
         COMMAND "${CMAKE_COMMAND}" -E copy_directory "${CMAKE_SOURCE_DIR}/bin" "$<TARGET_FILE_DIR:rpcs3>"
+        COMMAND "${CMAKE_COMMAND}" -E copy "${PROJECT_BINARY_DIR}/3rdparty/OpenAL/openal-soft/Debug/OpenAL32.dll" "$<TARGET_FILE_DIR:rpcs3>"
```

Comment:
This will only work with the Debug config and MSVC, but we support other configs and GCC, Clang on Windows.

Reply:
I can apply those changes only for MSVC if needed. The problem is that on Windows (at least on my 7800X3D) the `Release` mode had no effect at all. The application is always compiled in `Debug` mode from both VS and also standalone `cmake` (from command line).

Reply:
I don't think check_cxx_compiler_flag is called for MSVC.
I don't know anything about cmake, but this stood out to me.

Reply:
maybe it doesn't have the CMPXCHG16B instruction

Reply:
That's from the CMake log
```
        ClCompile:
          C:\\Program Files\\Microsoft Visual Studio\\2022\\Community\\VC\\Tools\\MSVC\\14.41.34120\\bin\\HostX64\\x64\\CL.exe /c /I"C:\\src\\vcpkg\\installed\\x64-windows\\include" /Zi /W1 /WX- /diagnostics:column /Od /Ob0 /D _MBCS /D WIN32 /D _WINDOWS /D COMPILER_X86 /D "CMAKE_INTDIR=\\"Debug\\"" /Gm- /EHsc /RTC1 /MDd /GS /fp:precise /Zc:wchar_t /Zc:forScope /Zc:inline /GR /std:c++20 /Fo"cmTC_a7f2c.dir\\Debug\\\\" /Fd"cmTC_a7f2c.dir\\Debug\\vc143.pdb" /external:W1 /Gd /TP /errorReport:queue  -msse -msse2 -mcx16 "E:\\build-rpcs3-msvc\\CMakeFiles\\CMakeScratch\\TryCompile-szusic\\src.cxx"
          Microsoft (R) C/C++ Optimizing Compiler Version 19.41.34120 for x64
          Copyright (C) Microsoft Corporation.  All rights reserved.
          cl /c /I"C:\\src\\vcpkg\\installed\\x64-windows\\include" /Zi /W1 /WX- /diagnostics:column /Od /Ob0 /D _MBCS /D WIN32 /D _WINDOWS /D COMPILER_X86 /D "CMAKE_INTDIR=\\"Debug\\"" /Gm- /EHsc /RTC1 /MDd /GS /fp:precise /Zc:wchar_t /Zc:forScope /Zc:inline /GR /std:c++20 /Fo"cmTC_a7f2c.dir\\Debug\\\\" /Fd"cmTC_a7f2c.dir\\Debug\\vc143.pdb" /external:W1 /Gd /TP /errorReport:queue  -msse -msse2 -mcx16 "E:\\build-rpcs3-msvc\\CMakeFiles\\CMakeScratch\\TryCompile-szusic\\src.cxx"
        cl : command line  warning D9002: ignoring unknown option '-msse' [E:\\build-rpcs3-msvc\\CMakeFiles\\CMakeScratch\\TryCompile-szusic\\cmTC_a7f2c.vcxproj]
        cl : command line  warning D9002: ignoring unknown option '-msse2' [E:\\build-rpcs3-msvc\\CMakeFiles\\CMakeScratch\\TryCompile-szusic\\cmTC_a7f2c.vcxproj]
        cl : command line  warning D9002: ignoring unknown option '-mcx16' [E:\\build-rpcs3-msvc\\CMakeFiles\\CMakeScratch\\TryCompile-szusic\\cmTC_a7f2c.vcxproj]
          src.cxx
```
MSVC literally does not no these flags because. Only GCC and Clang support those.

Reply:
I think you have to ifdef the check_cxx_compiler_flags to
`if(NOT WIN32 OR NOT MSVC)`
or something like that

Reply:
but did you (oltolm) have the issue I had in llvm's `CMakeLists.txt` (the linking error on LLVMIntelJITEvents module)? Which CPU do you have?

Reply:
Yes I had the same issue, but it was fixed by this change
```
		if(LLVM_USE_INTEL_JITEVENTS)
			list (APPEND LLVM_ADDITIONAL_LIBS IntelJITEvents)
		endif()
		if(LLVM_USE_PERF)
			list (APPEND LLVM_ADDITIONAL_LIBS PerfJITEvents)
		endif()
```
`COMPILER_X86` is not defined for VC. You can check it yourself with a CMake debugger or just by adding this to `CMakeLists.txt`
```
message("COMPILER_X86: ${COMPILER_X86}")
```

Reply:
> Yes I had the same issue, but it was fixed by this change
> 
> ```
> 		if(LLVM_USE_INTEL_JITEVENTS)
> 			list (APPEND LLVM_ADDITIONAL_LIBS IntelJITEvents)
> 		endif()
> 		if(LLVM_USE_PERF)
> 			list (APPEND LLVM_ADDITIONAL_LIBS PerfJITEvents)
> 		endif()
> ```
> 

yes I saw your other post. So with this solution we will link PerfJITEvents instead of IntelJITEvents. I don't know if that is ok or could be worse.

Reply:
fixed